### PR TITLE
Update state persistence docs

### PR DESCRIPTION
**Pull request by m-murasovs**
Preview/staging: https://docs.apify.com/actors/development/state-persistence?version=update/state-persistence

### CODE REVIEW

**Review by mnmkng — CHANGES_REQUESTED** [review]
Looks good to me besides a few bullet points:

- "check for persisted state" link does not scroll to the example
- not sure if we want to tell users about spot instances @jancurn 
- The message conveyed by `We have dataset, key-value store, and request queue storage so we could store results and in-progress data.` is not very clear.

**Comment by jancurn** [comment]
> * not sure if we want to tell users about spot instances @jancurn

Good point @mnmkng! Yeah, this is an internal detail that doesn't need to be communicated, it might only confuse people. We can just say that the jobs might be removed from the worker machines to optimize the workload or something like that.

File: docs/actors/development/state_persistence.md [hunk 1]
```diff
@@ -20,22 +20,20 @@ A migration is when a process running on a server has to stop and move to anothe
 
 **When a migration event occurs, you only have a few seconds to save your work.**
 
-## [](#why-migrations-happen)Why do migrations happen?
+## [](#why-do-migrations-happen)Why do migrations happen?
 
-Migrations happen because of the availability of Amazon EC2's **spot instances**. [Visit Amazon for more information](https://aws.amazon.com/ec2/spot/?cards.sort-by=item.additionalFields.startDateTime&cards.sort-order=asc).
+- To optimize server workloads.
+- When a server crashes (unlikely).
+- During deploys of new [worker](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers) software to the server.
```
Comment (mnmkng) [inline docs/actors/development/state_persistence.md:27]: It's actually [this worker](https://github.com/apify/apify-worker), not a web worker, that we deploy :) 

Maybe we can just say "When we release new features and fix bugs."

**Review by mnmkng — CHANGES_REQUESTED** [review]

File: docs/actors/development/state_persistence.md (hunk 1 above)
Comment (m-murasovs) [reply docs/actors/development/state_persistence.md:27]: Yep, that's way better :D

**Review by mnmkng — APPROVED** [review]

**Pull request merged**